src/fbvector/fbvector_formatter.py

```diff
@@ -0,0 +1,136 @@
+    try:
+        if not (valobj.IsValid()):
+            return "<invalid>"
+        
+        size = vec.num_children()
+        _summary += f"size={size}"
+
+        if size > 0:
+            _summary += " {\n"
+            for i in range(size):
+                _summary += f"  {vec.get_child_at_index(i)}\n"
+            _summary += "}"
+            
+        return _summary
+
+    except Exception:
+        # todo: change this
+        return f"size=-1"
+
+
+class FBVectorFormatter:
+    def __init__(self, valobj):
+        self.valobj = valobj
+        self.count = None
+        self.update()
+    
+    def num_children_impl(self):
+        try:
+            start_val = self.start.GetValueAsUnsigned(0)
+            finish_val = self.finish.GetValueAsUnsigned(0)
+            end_val = self.end.GetValueAsUnsigned(0)
+            # Before a vector has been constructed, it will contain bad values
+            # so we really need to be careful about the length we return since
+            # uninitialized data can cause us to return a huge number. We need
+            # to also check for any of the start, finish or end of storage values
+            # being zero (NULL). If any are, then this vector has not been
+            # initialized yet and we should return zero
+
+            # Make sure nothing is NULL
+            if start_val == 0 or finish_val == 0 or end_val == 0:
+                return 0
+
+            # Make sure start is less than finish
+            if start_val >= finish_val:
+                return 0
+
+            # Make sure finish is less than or equal to end of storage
+            if finish_val > end_val:
+                return 0
+
+            # if we have a struct (or other data type that the compiler pads to native word size)
+            # this check might fail, unless the sizeof() we get is itself incremented to take the
+            # padding bytes into account - on current clang it looks like
+            # this is the case
+            num_children = (finish_val - start_val)
+            if (num_children % self.data_size) != 0:
+                return 0
+            else:
+                num_children = num_children // self.data_size
+            return num_children
+        except Exception as e:
+            return 0
+
+    def num_children(self):
+        if self.count is None:
+            self.count = self.num_children_impl()
+        return self.count
+
+    def get_child_index(self, name):
+        try:
+            return int(name.lstrip('[').rstrip(']'))
+        except:
+            return -1
+
+    def get_child_at_index(self, index):
+        if index < 0:
+            return None
+        if index >= self.num_children():
+            return None
+        try:
+            # offset = index * self.data_size
+            # return self.start.CreateChildAtOffset(
+            #     '[' + str(index) + ']', offset, self.data_type)
+            return lldb.value(self.start)[int(index)].sbvalue
```

Comment:
why do you have that comment here? which version is used in c++ std's vector implementation?

Reply:
The commented out version is used in the std::vector implementation
The one I've got there is used in the range formatter, and it makes more sense to me.

Reply:
Sure, that's fine. Just finish all the tests in python and good to go!